evm/src/eNear/contracts/ENearProxy.sol

```diff
@@ -53,6 +60,26 @@ contract ENearProxy is UUPSUpgradeable, AccessControlUpgradeable, ICustomMinter
         eNear.transferToNear(amount, string(''));
     }
 
+    function finaliseNearToEthTransfer(
+        bytes memory proofData,
+        uint64 proofBlockHeight
+    ) external whenNotPaused(PAUSED_LEGACY_FIN_TRANSFER) {
+        require(
+            prover.proveOutcome(proofData, proofBlockHeight),
+            "Proof should be valid"
+        );
+
+        eNear.finaliseNearToEthTransfer(proofData, proofBlockHeight);
```

Comment:
Why do we need to add a separate method to this contract? Why can't we finalize transactions directly in eNear?